runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollPlugin.java

```diff
@@ -434,6 +452,38 @@ public void onItemContainerChanged(final ItemContainerChanged event)
 		}
 	}
 
+	private void setActiveClue(Integer itemId)
+	{
+		ClueScroll clueScroll = findClueScroll(itemId);
+
+		if (clueScroll != null)
+		{
+			clueItemId = itemId;
+			updateClue(clueScroll);
+		}
+	}
+
+	private Set<Integer> getClueItemIdsInInventory(ItemContainer inventoryContainer)
+	{
+		// Remove clues no longer in inventory
+		clueItemIdsInInventory.removeIf(clue -> !inventoryContainer.contains(clue));
+
+		// Add new clues
+		Item[] inventory = inventoryContainer.getItems();
+		Set<Integer> cluesFound = new HashSet<>();
+
+		for (Item item : inventory)
+		{
+			int invItemId = item.getId();
+			if (findClueScroll(invItemId) != null)
```

Comment:
This lookup is very inefficient. All I need to know is if the ItemID is a Clue Scroll

Reply:
Resolved while accepting that solution will populate clueItemIdsInInventory with Beginner, Master, Challenge, and Veos Clues. The null check on line 459 should cover these.